src/sardana/macroserver/scan/gscan.py

```diff
@@ -369,6 +371,38 @@ def __init__(self, macro, generator=None, moveables=[], env={},
         # ---------------------------------------------------------------------
         self._setupEnvironment(env)
 
+    def _check_moveables_limits(self):
+        for m in self._moveables:
+            config = PyTango.AttributeProxy(
```

Comment:
Not sure, but maybe we could avoid creating a new proxy here (I think this would save an extra connection to the Database)..
Could you please try something like: `m.moveable.getAttribute("Position").range` (at least this should work with Taurus4).
What do you think?